crates/table/src/table.rs

```diff
@@ -530,26 +534,53 @@ impl Table {
         Ok(())
     }
 
+    /// Finds the [`RowPointer`] to the row in `target_table` equal, if any,
+    /// to the row `needle_ptr` in `needle_table`,
+    /// by any unique index in `target_table`.
+    ///
+    /// # Safety
+    ///
+    /// `needle_table.is_row_present(needle_ptr)` must hold.
+    unsafe fn find_same_row_via_unique_index(
+        target_table: &Table,
+        needle_table: &Table,
+        needle_bs: &dyn BlobStore,
+        needle_ptr: RowPointer,
+    ) -> Option<RowPointer> {
+        // Find the smallest unique index.
+        let (cols, idx) = target_table
+            .indexes
+            .iter()
+            .find(|(_, idx)| idx.is_unique())
+            .expect("there is at least one unique index");
```

Comment:
```suggestion
            .expect("There should be at least one unique index");
```

Formatting per https://doc.rust-lang.org/std/result/enum.Result.html#method.expect .